UPGRADE.md

```diff
@@ -0,0 +1,148 @@
+  at /home/nathanielc/.cargo/registry/src/index.crates.io-6f17d22bba15001f/libp2p-swarm-0.44.2/src/lib.rs:367
+
+2024-07-02T15:04:55.732300Z  INFO ceramic_p2p::node: iroh-p2p peerid: 12D3KooWD2C2aWKagfowBN7LDkiXpXyMiJyN4FAuEptqZAQ2GgZp
+  at p2p/src/node.rs:168
+
+2024-07-02T15:04:55.733218Z  INFO ceramic_p2p::rpc: p2p rpc listening on: mem
+  at p2p/src/rpc.rs:434
+  in ceramic_p2p::rpc::new with addr=mem
+
+2024-07-02T15:04:55.733239Z  INFO ceramic_p2p::node: Listen addrs: ["/ip4/0.0.0.0/tcp/4001", "/ip4/0.0.0.0/udp/4001/quic-v1"]
+  at p2p/src/node.rs:249
+  in ceramic_p2p::node::run
+
+2024-07-02T15:04:55.733257Z  INFO ceramic_p2p::node: Local Peer ID: 12D3KooWD2C2aWKagfowBN7LDkiXpXyMiJyN4FAuEptqZAQ2GgZp
+  at p2p/src/node.rs:250
+  in ceramic_p2p::node::run
+```
+
+
+
+### Configure `js-ceramic` for `ceramic-one`
+
+Change the `ipfs` section in the `daemon.config.json` to use `remote` mode and point it at the running `ceramic-one` daemon.
+By default the `ceramic-one` daemon listens on port 5101, if you configured it differently make sure to match that configuration for `js-ceramic` as well.
+
+```json
+"ipfs": {
+    "mode": "remote",
+    "host": "http://localhost:5101"
+  },
+```
+
+
+### Start `js-ceramic` Node and Validate
+
+Now start the `js-ceramic` node with its new configuration and environment.
+Validate the process starts as before and that your application can communicate with it as needed.
+
+To validate that everything worked you can run any tests you may have against the node or perform a random sample query to ensure all data has migrated.
+This will depend on the specifics of your application.
+
+## Recovery
+
+At this point you should have a working upgraded `js-ceramic` node but sometimes things go wrong.
+The Ceramic node does some basic checks on startup to ensure its various data store are in sync, if these check fail the process with exit with an error message about missing data.
+In that case or any other failure mode follow these steps to return your node to its working state.
+
+1. Stop the `ceramic-one` and `js-ceramic` processes.
+2. Revert the configuration and environment changes to `js-ceramic`.
+3. Start `js-ceramic` process.
+
+These steps work because the existing data is not modified during the migration, instead it is only read and copied.
+
+## Live Upgrade Steps
+
+If you cannot tolerate a window of downtime of o your Ceramic node you can perform a two pass migration process to minimize the amount of time your node is offline.
+The basic strategy here is to run two nodes behind a proxy in an active/passive mode.
+
+These steps are very similar to the normal upgrade process so please review those steps first.
+The high level sequence is:
+
+1. Migrate the data from the active old node.
+2. Start new node as the passive with the migrated data.
+3. Validate new passive node is working
+4. Switch active role to new node and passive role to old node.
```

Comment:
this means there's a period of time where the node is live but missing data.  I'm not sure that really counts as a no-downtime migration.  Missing data could cause application-level logic issues.  At the very least I think we need to call this out loudly as a serious caveat for this migration approach.